### ISSUE TITLE
io.py relocates device-resident diagrams instead of refusing them; from_giotto misses D21's data term

### ISSUE THREAD

**Issue by sushovan4** [akritihq/akriti#46]
Opened as the condition of approving #43. @ADSilberman flagged four gaps in that PR's description, then checked them against the code and corrected himself to **two** — this records the two, in his words where they are better than mine.

## 1. `save()` silently relocates a device-resident diagram (§10.1, `N10.1-5`, `N10.1-6`)

The specification requires `save` to **refuse** a non-host-resident array, naming the device and the remedy:

> `save` MUST raise `ValueError` naming the device and the remedy — the caller's own `.cpu()`, `jax.device_get`, or `xp.from_dlpack` — rather than let the backend do it silently

`io.py` does the opposite. `_to_numpy` (`io.py:226`) falls back through `.detach()` and then `.cpu()` when `np.asarray` raises:

```python
detach = getattr(value, "detach", None)
...
cpu = getattr(detached, "cpu", None)
if callable(cpu):
    detached = cpu()
```

So a CUDA-resident diagram is moved off-device and written as though nothing happened. Worse than the "does not yet check" the PR description claimed: not a missing guard, an active one pointing the wrong way. It is §9's silent-wrongness category arriving through our own I/O path — the caller asked to save a GPU diagram, got a file, and was never told a transfer occurred.

## 2. `from_giotto` enforces D21's argument half and not its data half (§11, `N11-9`–`N11-11`)

§11's impossibility check has three terms. `from_giotto` checks the argument and not the data, so a diagram with `reduced_homology=False`, `infinity_values=inf` and **every H0 death finite** is accepted and recorded `essential_bars = "faithful"`.

That is precisely the false declaration §5.1 requires the argument to prevent: the caller said the essential class was kept, the data says otherwise, and the provenance records the caller's claim rather than the observable fact.

## Not gaps, checked and cleared

Two of the original four were wrong, and the check is worth recording so nobody re-files them:

- **§3.1's clamp target and `clamped_rows`.** `_clamp_i6` repairs with `xp.where(repair, births, deaths)` — `death := birth` exactly — and `_diagram_from_columns` writes `clamped_rows` on the path all five adapters take, including when it is `0`.
- **§11.2's determinism case.** Already asserts `ZipInfo.date_time` rather than inferring from two timed writes.

## Why this is an issue rather than a line in `tasks/`

The four-versus-two error came from `tasks/rfc-0001-review-progress.md` being stale: he wrote the PR description from it, and I reviewed the description without checking the code. Two people reading the same stale working document is not review. `tasks/` is being deleted and gitignored for that reason, and this is where the gap lives instead.

Blocks nothing. #43 merges with the specification ahead of the implementation in these two places, deliberately and recorded.
